packages/editor/src/plugins/spoiler/editor.tsx

```diff
@@ -1,48 +1,83 @@
 import { PluginToolbar } from '@editor/editor-ui/plugin-toolbar'
 import { PluginDefaultTools } from '@editor/editor-ui/plugin-toolbar/plugin-tool-menu/plugin-default-tools'
+import { TextEditorFormattingOption } from '@editor/editor-ui/plugin-toolbar/text-controls/types'
+import { selectIsFocused, useAppSelector } from '@editor/store'
 import { EditorPluginType } from '@editor/types/editor-plugin-type'
 import { useEditorStrings } from '@serlo/frontend/src/contexts/logged-in-data-context'
 import { cn } from '@serlo/frontend/src/helper/cn'
+import { useEffect } from 'react'
 
 import type { SpoilerProps } from '.'
 import { SpoilerRenderer } from './renderer'
 
+const titleFormattingOptions = [
+  TextEditorFormattingOption.code,
+  TextEditorFormattingOption.math,
+  TextEditorFormattingOption.richTextBold,
+  TextEditorFormattingOption.richTextItalic,
+]
+
 export function SpoilerEditor(props: SpoilerProps) {
   const { state, id, focused } = props
+  const { title, richTitle, content } = state
   const editorStrings = useEditorStrings()
+  const isTitleFocused = useAppSelector((state) =>
+    selectIsFocused(state, richTitle.defined ? richTitle.id : '')
+  )
```

Comment:
This might make the whole plugin rerender a lot. Maybe:

```suggestion
  const richTitleId = useMemo(() => {
    return richTitle.defined ? richTitle.id : ''
  }, [richTitle.defined, richTitle.id])
  const isTitleFocused = useAppSelector((state) =>
    selectIsFocused(state, richTitleId)
  )
```